src/models/clustering.jl

```diff
@@ -30,22 +30,22 @@ meta(AffinityPropagation,
 const AgglomerativeClustering_ = skcl(:AgglomerativeClustering)
 @sk_uns mutable struct AgglomerativeClustering <: MMI.Unsupervised
     n_clusters::Int     = 2::(_ ≥ 1)
-    affinity::String    = "euclidean"::(_ in ("euclidean", "l1", "l2", "manhattan", "cosine", "precomputed"))
+    metric::String      = "euclidean"::(_ in ("euclidean", "l1", "l2", "manhattan", "cosine", "precomputed"))
```

Comment:
@OkonSamuel It seems this change is not in your PR. Can you comment?

Reply:
It's not. I'm aware of this but choose to leave the `affinity` parameter till the next minor release of python sklearn.

Reply:
@OkonSamuel suggests leaving this as `affinity` and living with the dep warnings until sk-learn 1.4 is released.

Reply:
I reverted these changes. It is probably worth doing a different PR down the road to make these changes.